ckanext/harvest/harvesters/ckanharvester.py

```diff
@@ -35,21 +36,23 @@ def _get_search_api_offset(self):
         return '/api/%d/search' % self.api_version
 
     def _get_content(self, url):
-        http_request = urllib2.Request(
-            url = url,
-        )
+        http_request = urllib2.Request(url=url)
 
-        api_key = self.config.get('api_key',None)
+        api_key = self.config.get('api_key')
         if api_key:
-            http_request.add_header('Authorization',api_key)
+            http_request.add_header('Authorization', api_key)
 
         try:
             http_response = urllib2.urlopen(http_request)
+        except urllib2.HTTPError, e:
+            if e.getcode() == 404:
+                raise ContentNotFoundError('HTTP error: %s' % e.code)
+            else:
+                raise ContentFetchError('HTTP error: %s' % e.code)
         except urllib2.URLError, e:
-            raise ContentFetchError(
-                'Could not fetch url: %s, error: %s' %
-                (url, str(e))
-            )
+            raise ContentFetchError('URL error: %s' % e.reason)
+        except httplib.HTTPException, e:
```

Comment:
Yeah urllib2 can raise this. It is an annoyance of urllib2 - a bug I guess. See http://www.voidspace.org.uk/python/articles/urllib2.shtml#badstatusline-and-httpexception and https://stackoverflow.com/questions/666022/what-errors-exceptions-do-i-need-to-handle-with-urllib2-request-urlopen